corehq/apps/users/decorators.py

```diff
@@ -79,3 +79,13 @@ def _inner(request, domain, couch_user_id, *args, **kwargs):
         else:
             return domain_specific_login_redirect(request, domain)
     return _inner
+
+
+def ensure_active_user_by_username(username):
+    ccu = CommCareUser.get_by_username(username)
+    valid, message, error_code = True, None, None
+    if ccu and not ccu.is_active:
+        valid, message, error_code = False, 'User deactivated', 'mobile.app.translation.user.is.deactivated'
+    elif CommCareUser.get_deleted_by_username(username):
+        valid, message, error_code = False, 'User deleted', 'mobile.app.translation.user.is.deleted'
```

Comment:
@phillipm : i think `user.deactivated` and 'user.deleted` suffices.
For detailed messages how about:
`Your account has been deactivated, please contact your domain admin to reactivate` considering this would be shown to a mobile user as-is ?